server/util/uuid/uuid.go

```diff
@@ -43,3 +54,56 @@ func StringToBytes(text string) ([]byte, error) {
 	}
 	return uuidBytes, nil
 }
+
+func handleHostIDError(err error) {
+	hostIDError = err
+	fillHostId()
+}
+
+func fillHostId() {
+	id, err := guuid.NewRandom()
+	hostID = id.String()
+	if err != nil {
+		hostIDError = err
+		hostID = string(guuid.NodeID())
+	}
+}
+
+func GetHostID() (string, error) {
+	hostIDOnce.Do(
+		func() {
+			userConfigDir, err := os.UserConfigDir()
+			if err != nil {
+				handleHostIDError(err)
+				return
+			}
+			hostIDFilepath := path.Join(userConfigDir, "buildbuddy", hostIDFilename)
```

Comment:
nit: maybe make buildbuddy a const above and add a comment that that dir will be a subdir in the userconfig dir

Reply:
Done.